File: bench/timers/1.go
package main

import (
	"os"
	"strconv"
	"fmt"
	"time"
)

func main() {
	var n = 30000000
	var args = os.Args[1:]
	if len(args) >= 1 {
		n, _ = strconv.Atoi(args[0])
	}

	var start = time.Now()

	for i := 0; i < n; i++ {
		time.Now()
	}

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}

File: bench/timers/1.cpp
#include <iostream>
#include <chrono>
#include <unistd.h>

using namespace std;

int main(int argc, char const *argv[])
{
    int n = 30000000; // 1 second
	if (argc == 2)
	{
		n = atoi(argv[1]);
	}

    auto start = chrono::steady_clock::now();

    for (int i = 0; i < n; ++i)
    {
        chrono::steady_clock::now();
    }

    cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

    return 0;
}
